readme.md

```diff
@@ -226,11 +226,28 @@ Change the spinner indent.
 ### ora.promise(action, text)
 ### ora.promise(action, options)
 
-Starts a spinner for a promise. The spinner is stopped with `.succeed()` if the promise fulfills or with `.fail()` if it rejects. Returns the spinner instance.
+Starts a spinner for a function or a promise. The spinner is stopped with `.succeed()` if the promise fulfills or with `.fail()` if it rejects. Returns the promise.
 
 #### action
 
-Type: `Promise`
+Type: `Promise | ((spinner: ora.Ora) => Promise)`
+
+#### options
+
+Type: `object`
+
+All of the [options](#options) plus the following:
+
+##### successText
+Type: `string | ((result: T) => string)`
```

Comment:
For the functions, I used this as the pattern: https://github.com/sindresorhus/got/blob/main/documentation/5-https.md#checkserveridentity. Please, change it further if wanted to fit your README styling.